File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-typescript/composed/src/session-module.ts
import type {
  PresenceDeviceRequest,
  PresenceSnapshotView,
  ResumeSessionRequest,
  SessionResumeView,
} from './types.js';
import type { CrawChatSdkContext } from './sdk-context.js';

export class CrawChatSessionModule {
  constructor(private readonly context: CrawChatSdkContext) {}

  resume(body: ResumeSessionRequest): Promise<SessionResumeView> {
    return this.context.backendClient.session.resume(body);
  }

  disconnectDevice(body: PresenceDeviceRequest): Promise<PresenceSnapshotView> {
    return this.context.backendClient.session.disconnect(body);
  }
}

File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-flutter/composed/lib/src/session_module.dart
import 'package:backend_sdk/backend_sdk.dart';

import 'context.dart';

class CrawChatSessionModule {
  final CrawChatSdkContext context;

  CrawChatSessionModule(this.context);

  Future<SessionResumeView?> resume(ResumeSessionRequest body) {
    return context.backendClient.session.resume(body);
  }

  Future<PresenceSnapshotView?> disconnectDevice(PresenceDeviceRequest body) {
    return context.backendClient.session.disconnect(body);
  }
}
